sdk/storage/azure-storage-file/apireview/azure/storage/file/aio/directory_client_async.pyi

```diff
@@ -0,0 +1,38 @@
+# Stubs for azure.storage.file.aio.directory_client_async (Python 3)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from .._deserialize import deserialize_directory_properties
+from .._generated.aio import AzureFileStorage
+from .._generated.models import HandleItem, StorageErrorException
+from .._generated.version import VERSION
+from .._parser import _datetime_to_str, _get_file_permission
+from .._shared.base_client_async import AsyncStorageAccountHostsMixin
+from .._shared.parser import _str
+from .._shared.policies_async import ExponentialRetry
+from .._shared.request_handlers import add_metadata_headers
+from .._shared.response_handlers import process_storage_error, return_response_headers
+from ..directory_client import DirectoryClient as DirectoryClientBase
+from ..models import ContentSettings, DirectoryProperties, NTFSAttributes, ShareProperties
+from ._polling_async import CloseHandlesAsync
+from .file_client_async import FileClient
+from .models import DirectoryPropertiesPaged, HandlesPaged
+from azure.core.async_paging import AsyncItemPaged
+from typing import Any, Dict, Optional, Union
+
+class DirectoryClient(AsyncStorageAccountHostsMixin, DirectoryClientBase):
+    def __init__(self, directory_url: str, share: Optional[Union[str, ShareProperties]]=..., directory_path: Optional[str]=..., snapshot: Optional[Union[str, Dict[str, Any]]]=..., credential: Optional[Any]=..., loop: Any=..., **kwargs: Optional[Any]) -> None: ...
+    def get_file_client(self, file_name: str, **kwargs: Any) -> FileClient: ...
+    def get_subdirectory_client(self, directory_name: str, **kwargs: Any) -> DirectoryClient: ...
+    async def create_directory(self, metadata: Optional[Dict[str, str]]=..., timeout: Optional[int]=..., **kwargs: Optional[Any]) -> Dict[str, Any]: ...
+    async def delete_directory(self, timeout: Optional[int]=..., **kwargs: Any) -> None: ...
+    def list_directories_and_files(self, name_starts_with: Optional[str]=..., timeout: Optional[int]=..., **kwargs: Any) -> AsyncItemPaged: ...
+    def list_handles(self, recursive: bool=..., timeout: Optional[int]=..., **kwargs: Any) -> AsyncItemPaged: ...
+    async def close_handles(self, handle: Union[str, HandleItem]=..., recursive: bool=..., timeout: Optional[int]=..., **kwargs: Any) -> Any: ...
+    async def get_directory_properties(self, timeout: Optional[int]=..., **kwargs: Any) -> DirectoryProperties: ...
+    async def set_directory_metadata(self, metadata: Dict[str, Any], timeout: Optional[int]=..., **kwargs: Any) -> Dict[str, Any]: ...
+    async def set_http_headers(self, file_attributes: Union[str, NTFSAttributes]=..., file_creation_time: Union[str, datetime]=..., file_last_write_time: Union[str, datetime]=..., file_permission: Optional[str]=..., file_permission_key: Optional[str]=..., timeout: Optional[int]=..., **kwargs: Any) -> Dict[str, Any]: ...
```

Comment:
Don't know the service enough, but having a method called "http_headers" where all parameters are about "file_xxx" seems weird to me. "set_file_properties" maybe or something that remind the parameters?

Reply:
not a priority at this point of time.